### PR TITLE
Change how List Blobs Perf Creates Blobs

### PR DESCRIPTION
**Pull request by alzimmermsft**
# Description

Another attempt at removing the need for `Schedulers.boundedElastic()` in `AysncRestProxy`. The change stems from using logging in Reactor Netty to find that the same EPoll thread, in Linux, creates all connections to the Storage account. This may be from blob creation happening sequentially, making it parallel forces multiple EPoll threads to create the connections as multiple requests with happen at the same time making it impossible for one thread to create all connections.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: common/perf-test-core/src/main/java/com/azure/perf/test/core/PerfStressProgram.java [hunk 1]
```diff
@@ -18,26 +18,38 @@
...
+            completedOperations += test.getCompletedOperations();
+        }
+
+        return completedOperations;
     }
 
     private static double getOperationsPerSecond(PerfTestBase<?>[] tests) {
-        return IntStream.range(0, tests.length)
-            .mapToDouble(i -> tests[i].getCompletedOperations() / (((double) tests[i].lastCompletionNanoTime) / NANOSECONDS_PER_SECOND))
-            .sum();
```
Comment (alzimmermsft) [inline common/perf-test-core/src/main/java/com/azure/perf/test/core/PerfStressProgram.java:40]: This had a fun issue where `Double.NaN` being added resulted in the value being `Double.NaN`.

**Review by ibrahimrabab — APPROVED** [review]

File: common/perf-test-core/src/main/java/com/azure/perf/test/core/ApiPerfTestBase.java [hunk 2]
```diff
@@ -163,35 +164,45 @@ public Mono<Void> runAllAsync(long endNanoTime) {
...
-            .doOnNext(v -> {
-                completedOperations += v;
+        return Flux.generate(sink -> {
+                if (System.nanoTime() < endNanoTime) {
+                    sink.next(1);
+                } else {
+                    sink.complete();
+                }
+            })
+            .flatMap(ignored -> {
```
Comment (mikeharder) [inline common/perf-test-core/src/main/java/com/azure/perf/test/core/ApiPerfTestBase.java:174]: Why do we need to check if the timer expired twice?  The timer is approximate so running one extra operation should be fine.  Returning a result of "0" won't increase completedOperations, but it will update lastCompletionNanoTime, which seems wrong.

File: common/perf-test-core/src/main/java/com/azure/perf/test/core/PerfStressProgram.java [hunk 3]
```diff
@@ -18,26 +18,38 @@
...
 
 /**
  * Represents the main program class which reflectively runs and manages the performance tests.
  */
 public class PerfStressProgram {
     private static final int NANOSECONDS_PER_SECOND = 1_000_000_000;
 
-    private static int getCompletedOperations(PerfTestBase<?>[] tests) {
-        return Stream.of(tests).mapToInt(perfStressTest -> Long.valueOf(perfStressTest.getCompletedOperations()).intValue()).sum();
+    private static long getCompletedOperations(PerfTestBase<?>[] tests) {
```
Comment (mikeharder) [inline common/perf-test-core/src/main/java/com/azure/perf/test/core/PerfStressProgram.java:34]: Why was this changed?  To avoid using Reactor?

File: common/perf-test-core/src/main/java/com/azure/perf/test/core/PerfStressProgram.java [hunk 4]
```diff
@@ -18,26 +18,38 @@
...
+        long completedOperations = 0;
+        for (PerfTestBase<?> test : tests) {
+            completedOperations += test.getCompletedOperations();
+        }
+
+        return completedOperations;
     }
 
     private static double getOperationsPerSecond(PerfTestBase<?>[] tests) {
-        return IntStream.range(0, tests.length)
```
Comment (mikeharder) [inline common/perf-test-core/src/main/java/com/azure/perf/test/core/PerfStressProgram.java:38]: Why was this changed? To avoid using Reactor?

File: common/perf-test-core/src/main/java/com/azure/perf/test/core/PerfStressProgram.java [hunk 5]
```diff
@@ -236,10 +248,16 @@ public static void runTests(PerfTestBase<?>[] tests, boolean sync, int parallel,
         try {
             if (sync) {
                 ForkJoinPool forkJoinPool = new ForkJoinPool(parallel);
-                forkJoinPool.submit(() -> {
-                    IntStream.range(0, parallel).parallel().forEach(i -> tests[i].runAll(endNanoTime));
-                }).get();
+                List<Callable<Integer>> operations = new ArrayList<>(parallel);
```
Comment (mikeharder) [inline common/perf-test-core/src/main/java/com/azure/perf/test/core/PerfStressProgram.java:251]: Why was this changed?

File: common/perf-test-core/src/main/java/com/azure/perf/test/core/PerfStressProgram.java [hunk 6]
```diff
@@ -251,13 +269,13 @@ public static void runTests(PerfTestBase<?>[] tests, boolean sync, int parallel,
                 });
 
                 Flux.range(0, parallel)
-                    .parallel()
-                    .runOn(Schedulers.boundedElastic())
-                    .flatMap(i -> tests[i].runAllAsync(endNanoTime))
+                    .parallel(parallel)
+                    .runOn(Schedulers.parallel())
+                    .flatMap(i -> tests[i].runAllAsync(endNanoTime), false, Math.min(parallel, 1000 / parallel), 1)
```
Comment (mikeharder) [inline common/perf-test-core/src/main/java/com/azure/perf/test/core/PerfStressProgram.java:274]: I think the `Math.min(parallel, 1000 / parallel)` might be incorrect.

Comment (alzimmermsft) [reply common/perf-test-core/src/main/java/com/azure/perf/test/core/PerfStressProgram.java:274]: This will be removed in #32020

File: common/perf-test-core/src/main/java/com/azure/perf/test/core/PerfStressProgram.java (hunk 5 above)
Comment (alzimmermsft) [reply common/perf-test-core/src/main/java/com/azure/perf/test/core/PerfStressProgram.java:251]: This was changed as the previous implementation didn't behave as expected. What it did before was create a single task that then ran whatever limit `.parallel()` had for parallelization. It now runs `parallel` tasks concurrently as expected.

File: common/perf-test-core/src/main/java/com/azure/perf/test/core/PerfStressProgram.java (hunk 3 above)
Comment (alzimmermsft) [reply common/perf-test-core/src/main/java/com/azure/perf/test/core/PerfStressProgram.java:34]: This was using Java Streams before but wasn't needed and just a simplification of logic.

File: common/perf-test-core/src/main/java/com/azure/perf/test/core/PerfStressProgram.java (hunk 4 above)
Comment (alzimmermsft) [reply common/perf-test-core/src/main/java/com/azure/perf/test/core/PerfStressProgram.java:38]: This was using Java Streams before but wasn't needed and just a simplification of logic.

File: common/perf-test-core/src/main/java/com/azure/perf/test/core/ApiPerfTestBase.java (hunk 2 above)
Comment (alzimmermsft) [reply common/perf-test-core/src/main/java/com/azure/perf/test/core/ApiPerfTestBase.java:174]: I'll revert this in #32020

File: common/perf-test-core/src/main/java/com/azure/perf/test/core/PerfStressProgram.java (hunk 5 above)
Comment (mikeharder) [reply common/perf-test-core/src/main/java/com/azure/perf/test/core/PerfStressProgram.java:251]: Are you saying the old code didn't actually run the tasks in parallel?  Or was it limited to a parallelism that might have been lower than requested?  Because I though I verified the sync tests ran in parallel as expected with the old code, but we mostly run async tests so this might have been a bug.

Comment (alzimmermsft) [reply common/perf-test-core/src/main/java/com/azure/perf/test/core/PerfStressProgram.java:251]: The latter, it ran in parallel, but the parallelism level wasn't in our control but in the control of however `Stream.parallel` is handled by the JDK.
